src/access/profiling/scaling.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Functions to calculate metrics related to parallel scaling of applications."""
+
+import xarray as xr
+
+from access.profiling.metrics import ProfilingMetric
+
+
+def parallel_speedup(stats: xr.Dataset, metric: ProfilingMetric) -> xr.DataArray:
+    """Calculates the parallel speedup from scaling data.
+
+    Args:
+        stats (Dataset): Scaling data, stored as a xarray dataset.
+        metric (ProfilingMetric): Metric to use for the speedup calculation.
+
+    Returns:
+        DataArray: Parallel speedup.
+    Raises:
+        ValueError: If metric units are not time (e.g., seconds).
+    """
+    if stats[metric].pint.units != "second":
+        raise ValueError("Metric units must be time (e.g., seconds)!")
+    speedup = stats[metric].sel(ncpus=stats["ncpus"].min()) / stats[metric]
+    speedup.name = "speedup"
+    return speedup
+
+
+def parallel_efficiency(stats: xr.Dataset, metric: ProfilingMetric) -> xr.DataArray:
+    """Calculates the parallel efficiency from scaling data.
+
+    Args:
+        stats (Dataset): Scaling data, stored as a xarray dataset.
+        metric (ProfilingMetric): Metric to use for the efficiency calculation.
+
+    Returns:
+        DataArray: Parallel efficiency.
+    """
+    speedup = parallel_speedup(stats, metric)
+    eff = speedup / speedup.ncpus * stats["ncpus"].min()
```

Comment:
This logic looks confusing - better to write as `speedup * (stats["ncpu"].min()/speedup.ncpus)`

Related: why does speedup have an "ncpus" attribute?And how is that attribute populated? Basically, It would read better if "ncpus" values came from `stats` and only the `speedup` array itself came from `speedup`

Reply:
> This logic looks confusing - better to write as speedup * (stats["ncpu"].min()/speedup.ncpus)

I rewrote it as:
```python
speedup * (speedup.ncpus.min() / speedup.ncpus)
```
to be fully consistent.

> Related: why does speedup have an "ncpus" attribute?And how is that attribute populated? Basically, It would read better if "ncpus" values came from stats and only the speedup array itself came from speedup

`ncpus` is a coordinate of the `stats` dataset and is therefore inherited by the `speedup` data array. In xarray, coordinates can be accessed as a key-value pair, `speedup["ncpus"]`, or as an attribute` speedup.ncpus`. My understanding is that both are equivalent.

Reply:
Great! Yes, makes more sense when both `ncpus` come from the same variable.